packages/core/app-extensions/src/externalEvents/sagas.js

```diff
@@ -1,14 +1,51 @@
-import {takeEvery, call, all, select} from 'redux-saga/effects'
+import {takeEvery, call, all, select, put} from 'redux-saga/effects'
 
 import * as actions from './actions'
 import {invokeExternalEvent} from './externalEvents'
 
 export default function* sagas(configSelector) {
-  yield all([takeEvery(actions.FIRE_EXTERNAL_EVENT, fireExternalEvent, configSelector)])
+  yield all([
+    takeEvery(actions.FIRE_EXTERNAL_EVENT, fireExternalEvent, configSelector),
+    takeEvery(actions.FIRE_EXTERNAL_EVENT, fireMappedExternalEvent, configSelector),
+    takeEvery(actions.FIRE_STATE_CHANGE_EVENT, fireStateChangeEvent)
+  ])
+}
+
+export function* selectConfig(configSelector) {
+  const config = yield select(configSelector)
+  if (typeof config?.events === 'object') {
+    return config
+  }
+  return config ? {events: config} : config
```

Comment:
suggestion[non-blocking]:

my brain copes much better with (which should be the same?):

```
if (config) {
   return config.events ? config : {events: config}
}
return null
```